universe/Conditions.cpp

```diff
@@ -8386,18 +8386,19 @@ namespace {
         // distance to point from line from vector projection / dot products
         //.......O
         //      /|
-        //     / |
-        //    /  |d
-        //   /   |
-        //  /a___|___
+        //     / |   1a
+        //    /  |d  |
+        //   /   |   |
+        //  /a___|___|
         // 1         2
-        // (1O).(12) = |1O| |12| cos(a)
-        // d = |1O| cos(a) = (1O).(12) / |12|
-        // d = -(O1).(12 / |12|)
+        // (1O).(12) = |1O| |12| sin(a)
```

Comment:
Need some explanation here...  `A•B = |A||B|cos(α)` where **α** is the angle between the vectors **A** and **B**.

What is **1a** and how is it different from **O** and what does it have to do with **d** and why isn't it just **3** ?

In the older version, **O** is the location of the object, and **1** and **2** are the ends of a starlane that we want to calculate how far is away from **O** at the closest point to the line through those stars. It tries to use a vector projection, but is probably calculating the projection of **1->0** onto **1->2** rather than the perpendicular distance.

I think your calculation is OK, but it doesn't need the confusing **1a** in the comment...

rather maybe:
````
        // distance to point from line from vector projection / dot products
        //       O
        //      /|
        //     / |
        //    /  |d
        //   /   |
        //  /a___|___
        // 1         2
        // (1O)x(12) = |1O| |12| sin(a)
        // d = |1O| sin(a) = (1O)x(12) / |12|
        // d = (10)x(12 / |12|)
         constexpr float MIN_PERP_DIST = 20; // magic limit, in units of universe units (uu)

        float perp_dist = std::abs(v_o1_x*v_12_y - v_o1_y*v_12_x);
````

Reply:
I haven't done much geometry in the last 30 years. I had to figure out what a dot product was and now after checking for cross product I agree that your explanation is better. Will replace the comment.